core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -0,0 +1,186 @@
+import { Plugin, PluginKey, EditorState, Transaction } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { Subject } from 'rxjs';
+import tippy, { Instance, Props } from 'tippy.js';
+
+import { ComponentRef } from '@angular/core';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { Editor } from '@tiptap/core';
+import { BubbleMenuView } from '@tiptap/extension-bubble-menu';
+
+import { AIContentPromptComponent } from '../ai-content-prompt.component';
+import { AI_CONTENT_PROMPT_PLUGIN_KEY } from '../ai-content-prompt.extension';
+import { TIPPY_OPTIONS } from '../utils';
+
+interface AIContentPromptProps {
+    pluginKey: PluginKey;
+    editor: Editor;
+    element: HTMLElement;
+    tippyOptions?: Partial<Props>;
+    component: ComponentRef<AIContentPromptComponent>;
+}
+
+interface PluginState {
+    open: boolean;
+    form: [];
+}
+
+export type AIContentPromptViewProps = AIContentPromptProps & {
+    view: EditorView;
+};
+
+export class AIContentPromptView extends BubbleMenuView {
+    public editor: Editor;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component?: ComponentRef<AIContentPromptComponent>;
+
+    private $destroy = new Subject<boolean>();
```

Comment:
small typo here: `destroy$`

Reply:
@fmontes typo was "destroy$", I changed it to "$destroy"... this is the way you are defined in other places in the codebase

Reply:
Is actally `destroy$`